staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -113,6 +113,20 @@ type Config struct {
 	// Version forces a specific version to be used (if registered)
 	// Do we need this?
 	// Version string
+
+	// BearerTokenSource for managing the token
+	BearerTokenSource BearerTokenSource
```

Comment:
if the AuthProvider interface can return one of these, it seems odd to have AuthProviderConfig and a BearerTokenSource side by side in this struct